pilot/cmd/pilot-agent/main.go

```diff
@@ -562,6 +567,18 @@ func fromJSON(j string) *meshconfig.RemoteService {
 	return &m
 }
 
+func appendTLSCerts(rs *meshconfig.RemoteService) {
+	if rs.TlsSettings == nil {
+		return
+	}
+	if rs.TlsSettings.Mode == networking.TLSSettings_DISABLE {
+		return
+	}
+	rsTLSEnabled = true
```

Comment:
do you wnat to make sure that the values cacertificates, clientcertificate, etc. are not empty ? or does the code smartly ignore empty strings?

Reply:
I think it does not ignore empty. I thought since those are required fields inside `TLSSettings` we do not have worry about this case, but better to handle. I changed `dedupeStrings` to ignore empty strings. PTAL.